jira_utils.go

```diff
@@ -1,61 +1,59 @@
 package main
 
 import (
-	"bufio"
+	"encoding/json"
 	"fmt"
-	"os"
 	"strings"
-	"time"
 
 	bfconfluence "github.com/kentaro-m/blackfriday-confluence"
 	"github.com/michael-go/go-jsn/jsn"
 	bf "gopkg.in/russross/blackfriday.v2"
 )
 
-/***
-function getDate
-return date: string
-argument: none
-return a string containing date and time
-***/
-func getDate() string {
-
-	now := time.Now().Round(0)
-	y := fmt.Sprint(now.Year()) + "_"
-	m := fmt.Sprint(int(now.Month())) + "_"
-	d := fmt.Sprint(now.Day()) + "_"
-	h := fmt.Sprint(now.Hour()) + "_"
-	min := fmt.Sprint(now.Minute()) + "_"
-	s := fmt.Sprint(now.Second())
-
-	return y + m + d + h + min + s
+type JiraIssueForTicket struct {
+	Id  string `json:Id`
+	Key string `json:Key`
 }
 
-/***
-function findProjectId
-return found: bool
-return error
-input: projectId string
-input: filename path string
-return true if the project already exist in the file
-***/
-func findProjectId(projectId string, filename string, customDebug debug) (bool, error) {
-
-	f, err := os.Open(filename)
-	if err != nil {
-		customDebug.Debugf("*** ERROR *** couldn't find log file")
-		return false, err
-	}
-	defer f.Close()
+type JiradetailForTicket struct {
+	JiraIssue JiraIssueForTicket `json:jiraIssue`
+}
+
+type Tickets struct {
+	Summary         string `json:Summary`
+	Description     string `json:Description`
+	JiraIssueDetail map[string]interface{}
```

Comment:
I think an interface is still too generic here. Need something more defined, like another Struct defining the fields that go in JiraIssueDetail.

Reply:
the only way I can do that is if I decide not not put the issue ID in the JiraIssueDetail. Maybe I can do something like type Tickets struct {
	Summary         string `json:Summary`
	Description     string `json:Description`
        IssueId             string 'json:Id'
	JiraIssue        JiraIssue 'json:jiraIssue'
} with jira issue containing JiraIssue  ID and Key ?

Reply:
can't we have IssueId being optional?